File: code/functions/callbacks.go
package main

import "fmt"

// Callbacks in Go
func callback(value int) {
	fmt.Printf("Callback called with: %d\n", value)
}

func processWithCallback(cb func(int), data int) {
	cb(data)
}

func main() {
	processWithCallback(callback, 42)
}

File: code/functions/callbacks.scala
object Callbacks extends App {
  type Callback = Int =&gt; Unit
  
  def callback(value: Int): Unit = {
    println(s"Callback called with: $value")
  }
  
  def processWithCallback(cb: Callback, data: Int): Unit = {
    cb(data)
  }
  
  processWithCallback(callback, 42)
}